code/modules/modular_computers/file_system/programs/airestorer.dm

```diff
@@ -30,6 +30,11 @@
 	examine_text += span_info("Alt-click to eject the intelliCard.")
 	return examine_text
 
+/datum/computer_file/program/ai_restorer/Destroy(force)
+	if (stored_card) // The linter yells at me if I try to call try_eject here, so we have to do this instead
+		stored_card.forceMove(computer.drop_location())
+	return ..()
```

Comment:
`Destroy` should not have side effects, this should be happen in some kind of `deconstruct` proc

IE, when the computer it is hosted in is deconstructed (this is called when bashed to bits or by taken apart via tools), eject the ID card.

Reply:
Alright, all this logic is now handled in the `deconstruct` proc. I had to refactor it a little to make it work. I also took the liberty of ensuring that inserted PAIs would also be ejected when consoles are destroyed.